### DISCUSSION TITLE
ScaledLoss class: way to set the scale or to construct a new ScaledLoss from another instance and a scale

### DISCUSSION THREAD

**Discussion by ThomChap** [ceres-solver/ceres-solver#968] [Q&A] [answered]
Hi

I need to modify the scale in ScaledLoss (because I can't infer the right value when creating the instance and can only compute it later on).  
Could you please consider adding a setter for a_ in class ScaledLoss (and not keeping a_ as const anymore) ?
Alternatively, could you please consider adding a new constructor taking a const ScaledLoss& and the new scale as input?

(in version 2.0.0 I found a workaround by deriving from class ScaledLoss to store a duplicate member rho_, then used the constructor with rho_, and the new weight as input. But in more recent versions, class ScaledLoss has been set to 'final', so my workaround no longer works). 

Thanks,
Thomas

**Comment by sandwichmaker** [comment, marked as answer]
Yes thats a reasonable thing to do. Please send me a change following the instructions at 

http://ceres-solver.org/contributing.html
